packages/S11-modules/multi-version/lib/MultiVer.rakumod

```diff
@@ -0,0 +1,142 @@
+use v6;
+unit class MultiVer;
+use Test::Util;
+
+class Foo:ver<0.0.2> { }
+
+has $.prefix;
+has @.distributions;
+has $.standalone;
+has $.standalone-script;
+has $.inst-dir;
+has $.inst-repo;
+has $.use-lib;
+
+submethod TWEAK {
+    $!prefix //= make-temp-dir;
+}
+
+method mksubdir(IO:D() $path) {
+    my $fullpath = $path.is-relative ?? $!prefix.add($path) !! $path;
+    return $fullpath if $fullpath.e;
+    my @path = $*SPEC.splitdir: $fullpath.resolve.relative($!prefix).IO;
+    my $subdir = $!prefix;
+    for @path -> $dir {
+        die "Can only create dirs under \$.prefix, but '" ~ $path ~ "' is outside of it" if $dir eq '..';
+        $subdir .= add($dir);
+        next if $subdir.e;
+        $subdir.mkdir;
+    }
+    $subdir
+}
+
+method rmsubdir(IO:D() $entry) {
+    return unless $entry.e;
+    if $entry.d {
+        for $entry.dir -> $subentry {
+            next if $subentry.basename eq '..' | '.';
+            if $subentry.d {
+                self.rmsubdir($subentry);
+            }
+            else {
+                $subentry.unlink;
+            }
+        }
+        $entry.rmdir;
+    }
+    else {
+        $entry.unink;
+    }
+}
+
+method make-dist(Str:D :$dist = 'MVDist', :%fields, Bool :$standalone, Bool :$include = True) {
+    my %dist-fields =
+        api => "0.0.1",
+        ver => "0.0.1",
+        auth => 'test:raku',
+        description => 'Test distribution',
+        |%fields;
+
+    my $dest-dir = $.prefix.add($dist ~ "-" ~ %dist-fields<ver>).IO;
+
+    # Remove the previous installation to guarantee no garbage is left behind.
+    self.rmsubdir($dest-dir);
+
+    for $?DISTRIBUTION.meta<resources>.grep({ .starts-with($dist ~ '/') && .ends-with('.in') }) -> IO() $tmpl {
```

Comment:
The exposed public interface is actually `$*DISTRIBUTION`.I have no idea why I used `$?DISTRIBUTION` elsewhere... I think originally it *was* `$?DISTRIBUTION` but then later I changed it to `$*DISTRIBUTION` (which still requires setting up a `$*DISTRIBUTION`)

Reply:
Documentation mentions `$?DISTRIBUTION`, the compiler installs it too. And it's the right name for it. With `$*DISTRIBUTION` there is a chance to pick up a wrong one from the call stack. Yet, the lexical nature of `$?DISTRIBUTION` makes it more error-proof.

Contrary, `$*DISTRIBUTION` is good for the compilation.